src/transformers/generation_utils.py

```diff
@@ -1667,9 +1667,8 @@ def beam_search(
                 next_token_logits, cur_len=cur_len, max_length=max_length
             )
 
-            next_token_scores = F.log_softmax(next_token_logits, dim=-1)  # (batch_size * num_beams, vocab_size)
-
-            next_token_scores = logits_processor(input_ids, next_token_scores)
+            next_token_scores = logits_processor(input_ids, next_token_logits)
```

Comment:
I don't think we can do this change at all tbh. This is very backward's breaking and would mean that the beam search generation of everybody is affected.

I agree that the next_token_scores are not normalized anymore after having applied `logits_processor`, but it's not something we can change really anymore IMO since so many models rely on `beam_search`.

One thing we could do though to help your use case would maybe to add a new optional "Normalizer" `logits_processor` that would normalize the output after all other logits processors have been applied. Would that maybe help solve your use case?

Reply:
The Normalizer could be either another `log softmax` or just a `log x/Sum_i=1 y_i` normalized

Reply:
I actually implemented a workaround for my own code by doing that: putting a normalized logits processor at the end. So I don't need a change in this library. I understand this would break everybody's code, but the actual implementation of beam search is flawed. It's not just my use case but everybody's.

What if we add a flag to `generate` like `add_beam_search_normalization_fix` that optionally re-normalizes the scores after the logits processors? And then, when introducing more breaking changes (a major version change?), make it the default?

Reply:
I would prefer to just add a logits processor -> this would correct the behavior for everybody as well and is cleaner IMO

Reply:
Okay. I can give it a shot.